### PR TITLE
libappindicator: 12.10.0 -> 12.10.1.201020200706.1

### LINKED ISSUE #96063: discord: Frequent segmentation faults in libappindicator
**Describe the bug**
Discord crashes intermittently (maybe every 30 minutes) for me. This is especially noticeable when in a voice room, but I think it happens independently of that.

**To Reproduce**
1. Leave Discord running for a long period of time.
2. Observe that Discord segfaults.

**Additional context**
Stack trace in libappindicator-gtk3:
```
(gdb) bt
#0  0x00007f09f8fa015b in status_icon_changes () from /nix/store/s20y6yrgbrr1p5h70ydwim8z1q9niizk-libappindicator-gtk3-12.10.0/lib/libappindicator3.so
#1  0x00007f0a1ad94ec2 in g_closure_invoke () from /nix/store/mazsps24z01540n3g01m8yf52wm6avkd-glib-2.64.4/lib/libgobject-2.0.so.0
#2  0x00007f0a1ada7dc4 in signal_emit_unlocked_R () from /nix/store/mazsps24z01540n3g01m8yf52wm6avkd-glib-2.64.4/lib/libgobject-2.0.so.0
#3  0x00007f0a1adb2fef in g_signal_emit_valist () from /nix/store/mazsps24z01540n3g01m8yf52wm6avkd-glib-2.64.4/lib/libgobject-2.0.so.0
#4  0x00007f0a1adb397f in g_signal_emit () from /nix/store/mazsps24z01540n3g01m8yf52wm6avkd-glib-2.64.4/lib/libgobject-2.0.so.0
#5  0x00007f09f8f9f1d6 in app_indicator_set_icon_full () from /nix/store/s20y6yrgbrr1p5h70ydwim8z1q9niizk-libappindicator-gtk3-12.10.0/lib/libappindicator3.so
```

The rest of the stack has no symbols at all.

This has also been reported at https://www.reddit.com/r/linuxquestions/comments/fprclz/discord_segmentation_fault/ with the workaround being to remove libappindicator.

The crashes only started recently, presumably when #91375 landed.

**Notify maintainers**
@ldesgoui @MP2E @tadeokondrak @benley 

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.57, NixOS, 20.09pre237781.32b46dd897a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(goffrie): `"home-manager, nixpkgs-20.09pre238508.729e7295cf7"`
 - channels(root): `"nixos-20.09pre237781.32b46dd897a"`
 - nixpkgs: `/home/goffrie/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
another attempt to fix discord

closes #96246
closes #96063
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

failures broken on master
```
https://github.com/NixOS/nixpkgs/pull/96296
3 packages failed to build:
quodlibet-full quodlibet-xine-full steam-run-native

80 packages built:
Sylk appimage-run audio-recorder autokey blueman deltachat-electron devdocs-desktop discord discord-canary discord-ptb dropbox dropbox-cli gammastep gitter gnome3.gnome-control-center gnome3.networkmanagerapplet gromit-mpx gspeech gxneur handbrake hubstaff indicator-application-gtk2 indicator-application-gtk3 irccloud joplin-desktop kazam keybase-gui ledger-live-desktop libappindicator libappindicator-gtk2 marktext mate.caja-dropbox mate.mate-control-center mate.mate-polkit meteo minetime minishift modem-manager-gui molotov mullvad-vpn mycrypto notable obsidian pantheon.elementary-greeter pantheon.switchboard-plug-network pantheon.switchboard-with-plugs pantheon.wingpanel-with-indicators pasystray perl528Packages.Gtk2AppIndicator perl530Packages.Gtk2AppIndicator plexamp protonvpn-gui python27Packages.syncthing-gtk radiotray-ng redshift redshift-plasma-applet redshift-wlr remmina runwayml safeeyes sc-controller shutter signal-desktop skype slack ssb-patchwork standardnotes station syncthing-tray systrayhelper timeular tusk udiskie uget uget-integrator ulauncher unityhub wootility xflux-gui zulip
```

### CODE REVIEW

File: pkgs/development/libraries/libappindicator/default.nix [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
 
-  nativeBuildInputs = [ pkgconfig autoreconfHook vala gobject-introspection ];
+  sourceRoot = "~indicator-applet-developers/libappindicator/trunk";
+
+  nativeBuildInputs = [
+    pkg-config
+    autoreconfHook
+    vala
+    gobject-introspection
+    gnome2.gtkdoc # couldn't find a gnome3.gtkdoc related tool
```
Comment (worldofpeace) [inline pkgs/development/libraries/libappindicator/default.nix:35]: This is actually an alias for toplevel `gtk-doc`. not sure if it's just the gtk 3 version that needs this?

Comment (jonringer) [reply pkgs/development/libraries/libappindicator/default.nix:35]: both versions need it, but it's essentially just needed to call `gtkdocize` as part of the `autogen.sh` pre-configure

Comment (jonringer) [reply pkgs/development/libraries/libappindicator/default.nix]: I replaced it with the top-level attr

Comment (jonringer) [reply pkgs/development/libraries/libappindicator/default.nix]: without it, you will get:
```
checking for gtk-doc >= 1.0...
  testing gtkdocize... not found.
***Error***: You must have gtk-doc >= 1.0 installed
  to build libappindicator.  Download the appropriate package for
  from your distribution or get the source tarball at
    http://ftp.gnome.org/pub/GNOME/sources/gtk-doc/
```

**Comment by JJJollyjim** [comment]
Is this different from https://github.com/NixOS/nixpkgs/pull/93453 ?

**Comment by jonringer** [comment]
> Is this different from #93453 ?

not significantly, and I think I like there's more

**Pull request closed without merging**